### PR TITLE
Update Module Version Bump Action to target the new documentation

### LINKED ISSUE #1133: Update version bump workflow to work with new docs
# 🐛 Summary #

The version bump automation assumes a certain document structure when updating version references.  The upcoming #1114 doc restructuring adds new files with version references.  The workflow needs to be updated to bump those version references as well.

## To reproduce ##

1. Run version bump workflow on branch with new docs
1. Verify references to the old version still exist

## Expected behavior ##
All version references to be properly updated

### LINKED ISSUE #1183: Update run_module_version_bump.yml to include psgallery.md
# 🐛 Summary #

In [psgallery.md](https://github.com/cisagov/ScubaGear/blob/main/docs/installation/psgallery.md) we reference version 1.2.0, run_module_version_bump.yml should be updated to auto adjust to the latest version number here as well.

Screenshot (109)

## Expected behavior ##

run_module_version_bump.yml workflow should parse psgallery.md and auto adjust any references of the semver release number to the latest ScubaGear version.

### PR DESCRIPTION
**Pull request by buidav**
## 🗣 Description ##

- Update the Module Version Bump to change version numbers in the documentation structure
- Refactor out the Module Version bump GitHub action workflow code into a unit testable PowerShell module.

## 💭 Motivation and context ##

Closes #1183 
Closes #1133 

## 🧪 Testing ##

- Forked this branch and ran the workflow on the [fork](https://github.com/buidav/ScubaGear/pull/54) to bump the version.
- On a fork of the ScubaGear repo go to 
   - `Actions => Module Version Bump => Run Workflow => enter a semver version X.X.X version=> Run workflow`
   - To test the action

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [x] Resolved all merge conflicts on branch
- [x] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

**Comment by james-garriss** [comment]
I appreciate you refactoring the code to be unit testable.  You have earned this:

images

**Comment by james-garriss** [comment]
How do I test this code, @buidav ?  Should I actually run it, which would change things?

**Review by james-garriss — APPROVED** [review]
The good news is that I got this workflow to run in a fork of ScubaGear.  The bad news is that I think the input needs some validation.  As this is part of an internal process, I could be talked out of this.  Feel free to ping me w your thoughts.

File: utils/workflow/Set-ScubaGearModuleVersion.psm1 [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    # Replace the module version in the documentation
+    Set-ScubaGearVersionDoc
+
+    # Create the GitHub Pull Request
+    $PRTemplateContent = New-PRBody
+
+    # Create the PR
+    # TODO abstract this out into a reusable function in the future.
+    $ScubaGearVersionBumpBranch = "scubagear-version-bump-${env:NEW_VERSION_NUMBER}"
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply utils/workflow/Set-ScubaGearModuleVersion.psm1:194]: Well done.

**Review by dagarwal-mitre — APPROVED** [review]
Looks good!

**Pull request merged**